src/graph_world/nodeclassification/generator_wrappers.py

```diff
@@ -53,14 +64,36 @@ def Generate(self, sample_id):
         self._marginal)
     generator_config['generator_name'] = 'StochasticBlockModel'
 
+    if self._use_generated_lfr_communities: 
+      # Generating LFR graph to copy community sizes
+      _, lfr_model = SimulateLFRWrapper(generator_config['nvertex'], 
+                                        generator_config['avg_degree'],
+                                        int(generator_config['max_degree_proportion']*generator_config['nvertex']),
+                                        generator_config['power_exponent'],
+                                        int(generator_config['community_min_size_proportion']*generator_config['nvertex']), 
+                                        int(generator_config['community_max_size_proportion']*generator_config['nvertex']), 
+                                        generator_config['community_power_exponent'],
+                                        generator_config['mixing_param'])
+      if lfr_model is None:
+        return {'sample_id': sample_id,
+            'marginal_param': marginal_param,
+            'fixed_params': fixed_params,
+            'generator_config': generator_config,
+            'data': None}
+
+      communities = lfr_model.getPartition().subsetSizes()
+      generator_config['num_clusters'] = len(communities)
+      pi = np.array(communities) / sum(communities)
+      prop_mat = MakePropMat(generator_config['num_clusters'], generator_config['p_to_q_ratio'])
```

Comment:
This line is identical to the one in the `else` clause below. Please factor it out and add it outside the `if/else`, just before the `sbm_data` assignment.